src/modelframe.jl

```diff
@@ -57,12 +57,12 @@ function missing_omit(d::T) where T<:ColumnTable
         _nonmissing!(nonmissings, col)
     end
 
-    rows = findall(nonmissings)
+    rows = all(nonmissings) ? Colon() : findall(nonmissings)
     d_nonmissing =
-        NamedTuple{Tables.names(T)}(tuple((copyto!(similar(col,
-                                                           Base.nonmissingtype(eltype(col)),
-                                                           length(rows)),
-                                                   view(col, rows)) for col in d)...))
+        NamedTuple{Tables.names(T)}(tuple((colview = view(col, rows);
+                                           copyto!(similar(colview,
+                                                           Base.nonmissingtype(eltype(col))),
+                                                   colview)) for col in d)...)
```

Comment:
Benchmarking suggests that this is faster overall but especially when there are no missing values:
```julia
_missing_omit(x::AbstractVector{T}) where {T} = copyto!(similar(x, nonmissingtype(T)), x)
_missing_omit(x::AbstractVector, rows) = _missing_omit(view(x, rows))

function missing_omit(d::T) where T<:ColumnTable
    nonmissings = trues(length(first(d)))
    for col in d
        _nonmissing!(nonmissings, col)
    end
    d_nonmissing = if all(nonmissings)
        map(_missing_omit, d)
    else
        rows = findall(nonmissings)
        map(Base.Fix2(_missing_omit, rows), d)
    end
    d_nonmissing, nonmissings
end
```

Here's how I set up the benchmarking to compare:
```julia
julia> table1 = let rows = filter(!in((3, 9, 21, 22)), 1:25)
           (; (Symbol(x) => Union{Float64,Missing}[i in rows ? randn() : missing for i in 1:25]
               for x in 'a':'e')...)
       end;

julia> table2 = (; (Symbol(x) => randn(25) for x in 'a':'e')...);

julia> @benchmark missing_omit_pr($table1)
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.716 μs … 185.504 μs  ┊ GC (min … max): 0.00% … 96.95%
 Time  (median):     1.795 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.963 μs ±   3.538 μs  ┊ GC (mean ± σ):  3.52% ±  1.94%

  ▂▇█▇▅▃▄▄▄▃▃▂▁▁ ▁▁▁ ▁▁▁▁▂▂▃▂▁▁▁                              ▂
  ██████████████████████████████▇▆▅▅▄▁▄▃▁▃▁▄▄▄▁▁▃▁▁▄▁▃▁▃▁▄▄▄▅ █
  1.72 μs      Histogram: log(frequency) by time      3.19 μs <

 Memory estimate: 1.78 KiB, allocs estimate: 15.

julia> @benchmark missing_omit_pr($table2)
BenchmarkTools.Trial: 10000 samples with 10 evaluations.
 Range (min … max):  1.188 μs … 210.199 μs  ┊ GC (min … max): 0.00% … 97.53%
 Time  (median):     1.274 μs               ┊ GC (median):    0.00%
 Time  (mean ± σ):   1.418 μs ±   3.212 μs  ┊ GC (mean ± σ):  3.81% ±  1.69%

    █▇▄
  ▂████▆▄▄▃▃▃▃▃▃▂▂▃▃▂▃▃▃▄▃▃▃▃▂▂▂▁▂▂▂▂▂▂▂▂▂▁▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂▂ ▃
  1.19 μs         Histogram: frequency by time        2.42 μs <

 Memory estimate: 1.69 KiB, allocs estimate: 13.

julia> @benchmark missing_omit_suggestion($table1)
BenchmarkTools.Trial: 10000 samples with 97 evaluations.
 Range (min … max):  801.423 ns …  10.394 μs  ┊ GC (min … max): 0.00% … 87.02%
 Time  (median):     813.691 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   868.847 ns ± 468.232 ns  ┊ GC (mean ± σ):  2.79% ±  4.81%

  ▇█▃▃▁▁▄▂                        ▁                             ▁
  ████████▆▅▆▅▄▅▆█▇██▇▇▆▆▆█▄▅▅▇▅▄▇█▆▆▄▆█▅▆▅▅▆▅▅▅▅▄▄▃▄▃▃▃▃▂▄▄▄▄▄ █
  801 ns        Histogram: log(frequency) by time        1.4 μs <

 Memory estimate: 1.41 KiB, allocs estimate: 8.

julia> @benchmark missing_omit_suggestion($table2)
BenchmarkTools.Trial: 10000 samples with 194 evaluations.
 Range (min … max):  489.082 ns …   7.390 μs  ┊ GC (min … max): 0.00% … 89.22%
 Time  (median):     498.595 ns               ┊ GC (median):    0.00%
 Time  (mean ± σ):   552.261 ns ± 488.783 ns  ┊ GC (mean ± σ):  6.84% ±  7.13%

  ▆█▅▄▃▁▁   ▃▁  ▁                     ▁                         ▂
  ███████▇████▇█████▅▇▆▅▅▅▅▃▅▄▆▇▅▆▇▇▇██▇▆▇▇▇▅▅▄▃▄▄▅▃▄▄▁▄▄▃▄▄▄▁▃ █
  489 ns        Histogram: log(frequency) by time        868 ns <

 Memory estimate: 1.48 KiB, allocs estimate: 10.
```